tests/model_selection/scaffold_split.py

```diff
@@ -0,0 +1,167 @@
+from typing import Union
+
+import pytest
+from rdkit import Chem
+from rdkit.Chem import Mol
+
+from skfp.model_selection.scaffold_split import (
+    _create_scaffolds,
+    scaffold_train_test_split,
+    scaffold_train_valid_test_split,
+)
+
+
+@pytest.fixture
+def all_molecules() -> list[str]:
+    all_smiles: list[str] = [
+        "CCC",
+        "CCCl",
+        "CCBr",
+        "C1CC1",
+        "C1CCC1",
+        "C1CCCC1",
+        "C1CCCCC1",
+        "CCO",
+        "CCN",
+        "CC.OCC",
+    ]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in all_smiles]
+
+
+@pytest.fixture
+def disconnected_molecules() -> list[str]:
+    disconnected_smiles: list[str] = ["C1CC1.OCC", "C1CCC1.OCC", "CC.OCC"]
+
+    return [Chem.MolFromSmiles(smiles) for smiles in disconnected_smiles]
```

Comment:
If you are using a list in only one test, just make it there. Fixtures are only for data for multiple tests